src/main/java/io/cdap/plugin/gcp/bigtable/source/HBaseResultToRecordTransformer.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.gcp.bigtable.source;
+
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.format.UnexpectedFormatException;
+import io.cdap.cdap.api.data.schema.Schema;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.util.Bytes;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Transforms Google Cloud Bigtable {@link Result} to {@link StructuredRecord}.
+ */
+public class HBaseResultToRecordTransformer {
```

Comment:
please add unit tests for this class

Reply:
all unit and integration tests will be published in next PR for easier review